File: atcoder/ABC/162/C.py
import fractions
K = int(input())
sum = 0
D = {}
for a in range(1,K+1):
    for b in range(1,K+1):
        for c in range(1,K+1):
            if not (a+b+c in D) :
               D[a+b+c] = fractions.gcd(fractions.gcd(a,b),c)
            sum += D[a+b+c] 
print(sum)

File: atcoder/ABC/162/C.cpp
#include <bits/stdc++.h>
using namespace std;


int triple_gcd(int a, int b, int c){
    return gcd(gcd(a,b),c);
}

int main(){
    ios::sync_with_stdio(false);
    int N, sum = 0;
    cin >> N;
    for(int a = 1; a < N; a++){
        for(int b = 1; b < N; b++){
            for(int c = 1; c < N; c++){
                sum += triple_gcd(a, b, c);
            }
        }
    }
    cout << sum << endl;
}
